ext/pgsql/pgsql.c

```diff
@@ -1455,15 +1529,15 @@ PHP_FUNCTION(pg_last_notice)
 			break;
 		case PGSQL_NOTICE_ALL:
 			if (notices) {
-				RETURN_COPY(notices);
+				RETURN_ARR(zend_array_dup(notices));
 			} else {
 				array_init(return_value);
 				return;
 			}
 			break;
 		case PGSQL_NOTICE_CLEAR:
 			if (notices) {
-				zend_hash_clean(&PGG(notices));
```

Comment:
Didn't this remove all notices from all connections rather than just the notices for the current connection?